src/Analyzer/Resolve/QueryAnalyzer.cpp

```diff
@@ -310,12 +309,7 @@ std::optional<JoinTableSide> QueryAnalyzer::getColumnSideFromJoinTree(const Quer
         {
             auto table_side = getColumnSideFromJoinTree(argument_node, join_node);
             if (table_side && result && *table_side != *result)
-            {
-                throw Exception(ErrorCodes::AMBIGUOUS_IDENTIFIER,
-                    "Ambiguous identifier {}. In scope {}",
-                    resolved_identifier->formatASTForErrorMessage(),
-                    join_node.formatASTForErrorMessage());
-            }
+                return {};
```

Comment:
I can hardly imagine in which case we could end up here before.
Now, it's firstTruthy?
Does it make sense still throw an exception in other cases?